python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -16,58 +16,101 @@ class DataFlowCfgNode extends ControlFlowNode {
   DataFlowCfgNode() { isExpressionNode(this) }
 }
 
-/** A data flow node which should have an associated post-update node. */
-abstract class PreUpdateNode extends Node { }
+/** A data flow node for which we should synthesise an associated pre-update node. */
+abstract class NeedsSyntheticPreUpdateNode extends Node {
+  /** This will figure in the texttual representation of the synthesised pre-update node. */
```

Comment:
```suggestion
  /** A label for this kind of node. This will figure in the textual representation of the synthesized pre-update node. */
```
I'm still a bit meh on the name `label`, as it's awfully generic. Would something like `nodeKind` be better, I wonder? Or maybe `nodeKindLabel`?

Reply:
Or `preUpdateLabel`?